fix: issue warning when default aws region is used

fvdnabee opened the pull request:
### Description
Issue a warning to inform the user that (s)he forget to set `AWS_REGION` and that the default region will be used.
Note this PR is blocked by PR #151.

**Fixes:** https://github.com/IBM/argocd-vault-plugin/issues/154

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Docs have been added / updated

### Type of Change
<!-- Please check the one that applies to this PR using "x". -->
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### Other information
<!-- Please add any additional information that would help reviewers evaluate your PR -->

fvdnabee commented on pkg/config/config.go:
```diff
@@ -109,16 +109,18 @@ func New(v *viper.Viper, co *Options) (*Config, error) {
+				v.Set(types.EnvAWSRegion, types.AwsDefaultRegion)
 			}
-			s := session.Must(session.NewSession(&aws.Config{
+
```
Correct. 

Note this change is related to PR #151 and I commented briefly on it here: https://github.com/IBM/argocd-vault-plugin/pull/151#issuecomment-863888519 As I was unable to set the base of this PR to the last commit of PR #151, all of #151's commits show up in this PR as well. When reviewing this PR you should only review the last commit on this branch (97460cb).

jkayani reviewed (approved):
I'm good with this, just want to be sure that warning log message gets shown

werne2j commented:
@fvdnabee Can you rebase with Main and resolve the conflicts

fvdnabee commented:
> @fvdnabee Can you rebase with Main and resolve the conflicts

I rebased and added a test case verifying that the expected warning is outputted (1d03b53).

werne2j commented on pkg/config/config.go:
```diff
@@ -109,16 +109,18 @@ func New(v *viper.Viper, co *Options) (*Config, error) {
-					types.AWSSecretsManagerbackend,
-				)
+			if !v.IsSet(types.EnvAWSRegion) { // issue warning when using default region
+				log.Printf("Warning: %s env var not set, using AWS region %s.\n", types.EnvAWSRegion, types.AwsDefaultRegion)
```
Where did you confirm that this was going to `stderr`?

werne2j replied:
Ah, nevermind, found it `var std = New(os.Stderr, "", LstdFlags)`
